ls1intum/Artemis
# `Iris`: Change Pyris response type

### Checklist
#### General


#### Server
#### Client
No client changes

### Motivation and Context


Up until now, Pyris and Artemis have been unnecessarily coupled, in that Pyris' responses were always of the type `IrisMessage`, and Artemis would always deserialize them as such. However, this is an inflexible approach, as it only allows for one single AI-generated variable to be returned per call to Pyris, always the one named "response". This is inflexible. Guidance templates can have an arbitrary number of generations, and it is desirable behavior that you are also able to receive all of these generated values back from Pyris. Therefore I have adapted both Pyris and Artemis to communicate in more general types, while maintaining the current use-case of only having one response.

**Note: This change depends on Pyris sending the updated response type as well! This will not work with the current deployment of Pyris.**
**See corresponding Pyris PR: https://github.com/ls1intum/Pyris/pull/34**

### Description
- Updated the structure of `IrisMessageResponseDTO`. Instead of an `IrisMessage` it now contains a `JsonNode content` which represents an arbitrary content response from Pyris, for instance a list of JSON objects (which is very possible to generate with a guidance template).
- Added a method in `IrisChatSessionService` and `IrisHestiaSessionService` to reconstruct this arbitrary JSON into an `IrisMessage` for the existing use-case. The method for Hestia is identical, but @Hialus has said this is dead code anyway and will be changed in the future in one of his PRs. For now it is just there for compilation.
- Adapted the test cases correspondingly

### Steps for Testing

Prerequisites:
- **Pyris must be deployed on the branch of this PR: https://github.com/ls1intum/Pyris/pull/34**
- 1 Student
- 1 Programming Exercise with Iris enabled

1. Log in to Artemis
2. Navigate to the exercise
3. Make sure Iris still responds like normal to your chat input

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots
